lib/node/Helper.js

```diff
@@ -0,0 +1,78 @@
+const Negative = require('../Negative');
+const NodeCreator = require('./Creator');
+const NodeType = require('./Type');
+
+const NodeHelper = {};
+
+// Returns true if `node` belongs to the type specified by boolean `isTypeFunc`.
+// If `allowUnaryMinus/allowParens` is true, we allow for the node to be nested.
+NodeHelper.isType = function(node, isTypeFunc, allowUnaryMinus=true, allowParens=true) {
+  if (isTypeFunc(node)) {
+    return true;
+  }
+  else if (allowUnaryMinus && NodeType.isUnaryMinus(node)) {
+    return NodeHelper.isType(node.args[0], isTypeFunc, allowUnaryMinus, allowParens);
+  }
+  else if (allowParens && NodeType.isParenthesis(node)) {
+    return NodeHelper.isType(node.content, isTypeFunc, allowUnaryMinus, allowParens);
+  }
+
+  return false;
+};
+
+// Returns `node` if `node` belongs to the type specified by boolean `isTypeFunc`.
+// If `allowUnaryMinus/allowParens` is true, we check for an inner node of this type.
+// `moveUnaryMinus` should be defined if `allowUnaryMinus` is true, and should
+// move the unaryMinus into the inside of the type
+// e.g. for fractions, this function will negate the numerator
+NodeHelper.getType = function(
+  node, isTypeFunc, allowUnaryMinus=true, allowParens=true, moveUnaryMinus=undefined) {
+  if (allowUnaryMinus === true && moveUnaryMinus === undefined) {
+    throw Error('Error in `getType`: moveUnaryMinus is undefined');
+  }
+
+  if (isTypeFunc(node)) {
+    return node;
+  }
+  else if (allowUnaryMinus && NodeType.isUnaryMinus(node)) {
+    return moveUnaryMinus(
```

Comment:
Oooo right cause you have to define this - interesting! I wonder if we could default this to something - will be interesting to see if/how we find other cases for this. May as well leave as is for now